packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -119,42 +141,54 @@ export const useGridVirtualScroller = () => {
         apiRef.current.publishEvent('renderedRowsIntervalChange', nextRenderContext);
       }
 
+      previousScroll.current = scrollPosition.current;
       previousContext.current = rawRenderContext;
       prevTotalWidth.current = dimensions.columnsTotalWidth;
     },
     [apiRef, dimensions.isReady, dimensions.columnsTotalWidth],
   );
 
   const triggerUpdateRenderContext = () => {
-    const inputs = inputsSelector(apiRef, rootProps, enabled, enabledForColumns);
-    const [nextRenderContext, rawRenderContext] = computeRenderContext(inputs, scrollPosition);
+    const timestamp = performance.now();
+    const dt = timestamp - scrollCache.lastTimestamp;
+    const isScrolling = scrollCache.lastTimestamp !== -1 && dt < 500;
+
+    scrollCache.lastTimestamp = timestamp;
+    scrollCache.lastPosition = scrollPosition.current;
+
+    if (isScrolling) {
+      const dy = scrollPosition.current.top - previousScroll.current.top;
+      const dx = scrollPosition.current.left - previousScroll.current.left;
+      scrollCache.buffer = {
+        top: dimensions.rowHeight * 10,
+        left: 50 * 3,
+      };
```

Comment:
I tried predicting* the optimal buffer size but failed, so I just put a static 10 rows or 3 columns buffer.

*I had code in place to collect scroll points as event, which I hoped we could run some math on to figure out where the next scroll event would be. I tried linear regression, polynomial regression, and even a small (<1kb) ML model (simple perceptron architecture) with synaptic.js, but the problem is the data. One would expect scroll events to mostly fit to a bell curve. Which they do as a whole. But the intervals and delta values are just too irregular from one point to the next to be able to predict the next point in a sequence. I have gathered some data (me scrolling the grid) and [plotted it here](https://observablehq.com/d/bf8a4012ffdb8453), the last graph is an example of "generally like a bell curve but unpredictable from one point to the next". Maybe with `wheel` events added in the mix we could get a better error rate (best I could get was 50% error on average), but that will take some more research.

Reply:
I can't access the link attached. Could you update the permissions?

Reply:
Done